controllers/db/reports/publication.report.controller.ts

```diff
@@ -0,0 +1,754 @@
+          {
+            model: models.PersonPersonType,
+            as: "PersonPersonTypes",
+            required: isPersonTypeFilter,
+            on: {
+              col1: Sequelize.where(
+                Sequelize.col("AnalysisSummaryAuthor.personIdentifier"),
+                "=",
+                Sequelize.col("PersonPersonTypes.personIdentifier"),
+              ),
+              col2: Sequelize.where(
+                Sequelize.col("Person.personIdentifier"),
+                "=",
+                Sequelize.col("PersonPersonTypes.personIdentifier"),
+              ),
+            },
+            attributes: [[Sequelize.fn("GROUP_CONCAT", Sequelize.col('PersonPersonTypes.personType')),"personType",]],
+           // order: [],
+          }
+        ],
+        where: where,
+        group: ["AnalysisSummaryAuthor.pmid", "AnalysisSummaryAuthor.personIdentifier"],
+        order:[[ models.AnalysisSummaryArticle, 'datePublicationAddedToEntrez', 'DESC' ],
+               [models.PersonPersonType,'personType', 'ASC' ]],
+        subQuery: false,
+        attributes: []
+        // attributes:[[Sequelize.literal('DISTINCT "pmid"'),'pmid']]
+      })
+
+
+      console.log("result))))))", searchOutput)
+
+
+     /* let wherePersonIdentifier = [];
+      wherePersonIdentifier = searchOutput.map((data)=> data.Person.dataValues.personIdentifier)
+
+        const whereAuthors = {};
+        whereAuthors[Op.and] = [];
+        if (
+          apiBody.filters.personTypes &&
+          apiBody.filters.personTypes.length > 0
+        ) { 
+
+          isPersonType = true;
+          whereAuthors[Op.and].push({
+            "$PersonPersonType.personType$": {
+              [Op.in]: apiBody.filters.personTypes,
+            },
+          });
+          console.log("personIdentifierpersonIdentifierpersonIdentifierpersonIdentifier))))))")
+          whereAuthors[Op.and].push({
+            "$PersonPersonType.personIdentifier$": {
+              [Op.in]: wherePersonIdentifier,
+            },
+          });
+        }
+     /* personTypesOutput = await models.PersonPersonType.findAll({ 
+            where:  whereAuthors,
+            attributes: [[Sequelize.fn("GROUP_CONCAT", Sequelize.col('PersonPersonType.personType')),"personType"]],
+            group:["personIdentifier"],
+            order:[['personType', 'ASC']],
+           // attributes:['personIdentifier','personType"]
+      })
+
+      console.log("personTypesOutput##########",personTypesOutput);
+      if(personTypesOutput && personTypesOutput.length > 0 && wherePersonIdentifier && wherePersonIdentifier.length > 0)
+      {
+        
+        for(let i=0; i<wherePersonIdentifier.length;i++)
+        {
+            if(personTypesOutput.some(data => data.dataValues.personIdentifier === wherePersonIdentifier[i]))
+            {
+              searchOutput.push(personTypesOutput)
+            }
+
+        }
+
+      }*/
+      return searchOutput;
+    } catch (e) {
+      console.log(e);
```

Comment:
This function is pretty big. Would of been easier if this was broken up a bit

Reply:
Will refactor in the future releases.